File: platform/darwin/ios/lynx_devtool/lynx_recorder/LynxRecorderEnv.m
// Copyright 2025 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#import <LynxDevtool/LynxRecorderEnv.h>

@interface LynxRecorderEnv ()

@end

@implementation LynxRecorderEnv

+ (instancetype)sharedInstance {
  static LynxRecorderEnv *sharedInstance = nil;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^{
    sharedInstance = [[self alloc] init];
  });
  return sharedInstance;
}

- (instancetype)init {
  if (self = [super init]) {
    self.lynxRecorderUrlPrefix = @"file://lynxrecorder?";
    self.lynxRecorderUrlSchema = @"file";
    self.lynxRecorderUrlHost = @"lynxrecorder";
  }
  return self;
}
@end

File: platform/android/lynx_devtool/src/main/java/com/lynx/devtool/recorder/LynxRecorderEnv.java
// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

package com.lynx.devtool.recorder;

public class LynxRecorderEnv {
  private static LynxRecorderEnv instance;
  public String mUriKey = "uri";
  public String lynxRecorderUrlPrefix = "file://lynxrecorder?";

  public static synchronized LynxRecorderEnv getInstance() {
    if (instance == null) {
      instance = new LynxRecorderEnv();
    }
    return instance;
  }
}
